src/include/mpir_thread.h

```diff
@@ -95,6 +95,7 @@ extern MPID_Thread_mutex_t MPIR_THREAD_VCI_HANDLE_MUTEX;
 extern MPID_Thread_mutex_t MPIR_THREAD_VCI_CTX_MUTEX;
 extern MPID_Thread_mutex_t MPIR_THREAD_VCI_PMI_MUTEX;
 extern MPID_Thread_mutex_t MPIR_THREAD_VCI_BSEND_MUTEX;
+#define MPIR_THREAD_VCI_COMM_MUTEX(_comm_ptr) _comm_ptr->mutex
```

Comment:
There is no benefit of this macro, suggest remove. Just use `comm_ptr->mutex` directly.

Reply:
Removing it may be an option. I think the original motivation was to make it consistent with `MPIR_THREAD_POBJ_COMM_MUTEX` and `MPIR_THREAD_POBJ_WIN_MUTEX`.

Reply:
We have removed POBJ from ch4 layer completely.

Reply:
Let us move this discussion to the `allow_overtaking` PR https://github.com/pmodels/mpich/pull/5183.